source/type/type.cpp

```diff
@@ -423,7 +429,10 @@ VJ* MetaVehicleJourney::impl_create_vj(const std::string& uri,
     vj_ptr->realtime_level = level;
     vj_ptr->shift = 0;
     if (!sts.empty()) {
-        vj_ptr->shift = sts.front().arrival_time / (ndtu::SECONDS_PER_DAY);
+        const auto first_st_it = std::min_element(sts.begin(), sts.end(), [](const StopTime& st1, const StopTime& st2) {
```

Comment:
don't you have a `earliest_stop_time()` already? you can make the other one static